verifact/forum/models.py

```diff
@@ -1,15 +1,28 @@
 from django.db import models
 from django.utils.translation import gettext_lazy as _
+from django.contrib.contenttypes.fields import GenericForeignKey, GenericRelation
+from django.contrib.contenttypes.models import ContentType
+
+class Citation(models.Model):
+    created_at = models.DateTimeField(auto_now_add=True)
+    user = models.ForeignKey("auth.User", on_delete=models.CASCADE, related_name="citations")
+    citation_url = models.CharField(max_length=2048)
+    citation_title = models.CharField(max_length=2048)
+    citation_image_url = models.CharField(max_length=2048)
+
+    parent_type = models.ForeignKey(ContentType, on_delete=models.CASCADE)
```

Comment:
You prefer `parent_type` to `content_type`?

Reply:
You might want to limit model choices to `Question` and `Answer` to prevent weird relationships, like described here: https://www.javaer101.com/en/article/33695947.html